spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/SpanNameUtil.java

```diff
@@ -20,21 +20,38 @@
  * Utility class that provides the name in hyphen based notation
  *
  * @author Adrian Cole
- *
  * @since 1.0.2
  */
 public final class SpanNameUtil {
 
 	public static String toLowerHyphen(String name) {
 		StringBuilder result = new StringBuilder();
-		for (int i = 0; i < name.length(); i++) {
+		appendFirstChar(name, result);
+		for (int i = 1; i < name.length(); i++) {
 			char c = name.charAt(i);
-			if (c >= 'A' && c <= 'Z') {
+			if (isUppercase(c)) {
 				result.append('-').append((char) (c + 'a' - 'A'));
```

Comment:
looking at all the changes here.. in hindsight, I'd special case. Following advice is just what I'm thinking, not what you should do. You can leave the code as is, etc.

ex from the original code, replace

```java
result.append('-').append((char) (c + 'a' - 'A'));
```

with

```java
if (i != 0) result.append('-'); // special-case first character
result.append((char) (c + 'a' - 'A'));
```

That's only a 2 line change (net one additional line) as opposed to a 23 line change. In my mind at least, it accomplishes the same goal.